### PR TITLE
Integrate map window variables

### LINKED ISSUE #4345: x64 only: crash when opening map window
**OS:** Windows 10
**Version:** 0.0.5
**Commit/Build:** 89c9cc2

**Steps to reproduce:**
1. Open the map window.
2. Crash.

**Dump file**
[OpenRCT2.zip](https://github.com/OpenRCT2/OpenRCT2/files/441236/OpenRCT2.zip)

### PR DESCRIPTION
**Pull request by marijnvdwerf**
Should fix #4345

### CODE REVIEW

File: src/windows/map.c [hunk 1]
```diff
@@ -162,6 +162,14 @@ static rct_window_event_list window_map_events = {
 	window_map_scrollpaint
 };
 
+/** rct2: 0x00F1AD61 */
+static uint8 _activeTool;
```
Comment (marijnvdwerf) [inline src/windows/map.c]: 'Tool' probably isn't the best name

File: src/windows/map.c [hunk 2]
```diff
@@ -162,6 +162,14 @@ static rct_window_event_list window_map_events = {
...
+
+/** rct2: 0x00F1AD6C */
+static uint32 _curPixel;
```
Comment (marijnvdwerf) [inline src/windows/map.c]: `_currentLine`? I don't really get what this is doing.

File: src/windows/map.c [hunk 3]
```diff
@@ -162,6 +162,14 @@ static rct_window_event_list window_map_events = {
...
+
+/** rct2: 0x00F1AD68 */
+static uint8 _mapImageData[512][512];
```
Comment (marijnvdwerf) [inline src/windows/map.c]: Removed the dynamic allocation. Not sure if that's bad.

File: src/windows/map.c [hunk 4]
```diff
@@ -1674,15 +1678,19 @@ static void map_window_set_pixels(rct_window *w)
 				colour = map_window_get_pixel_colour_ride(x, y);
 				break;
 			}
-			*destination = colour;
+			destination[0] = HIBYTE(colour);
```
Comment (marijnvdwerf) [inline src/windows/map.c]: Now sure whether HI/LO are in the right order.

File: src/windows/map.c (hunk 3 above)
Comment (IntelOrca) [inline src/windows/map.c]: Well its unused memory when the window isn't open.

File: src/windows/map.c (hunk 2 above)
Comment (IntelOrca) [inline src/windows/map.c]: I think its current line. It draws it over multiple game ticks so it stores it statically.

**Comment by marijnvdwerf** [comment]
@IntelOrca I tried to make it use a pointer. It compiles and the map renders, but would be nice if you could check the changes

**Pull request merged**